### PR TITLE
Add useChainConnectionState hook

### PR DESCRIPTION
**Pull request by DoubleOTheven**
* Add useChainConnectionState hook
* Add example in Getting Started

I made this for my project tonight and found it helpful. Added it here to share. Example Use Case:

```js
  export function App() {
    const { activateBrowserWallet } = useEthers()
    const ETHEREUM_MAINNET_ID = 1;

    // the parameter is optional for useChainConnectionState. isWrongNetwork can only ever be true you pass
    // an ID in and the actual network ID is not undefined and does not match your expected ID.
    const { isDisconnected, isPendingNetworkConnection, isWrongNetwork, isConnected } = useChainConnectionState(ETHEREUM_MAINNET_ID);

    if (isDisconnected) {
      return (
        <div>
          <h1>Please connect your wallet</h1>
          <button onClick={activateBrowserWallet}>Connect</button>
        </div>
      )
    }

    if (isPendingNetworkConnection) {
      return (
        <div>
          <h1>loading...</h1>
        </div>
      )
    }

    if (isWrongNetwork) {
      return (
        <div>
          <h1>Your are connected, but you are not on Ethereum Mainnet!</h1>
        </div>
      )
    }

    return (
      <div>
        <h1>GM. You are connected and on Ethereum Mainnet!</h1>
      </div>
    )
  }
  ```

### CODE REVIEW

File: packages/core/src/hooks/index.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 export * from './useContractCall'
 export * from './useContractFunction'
+export * from './useDebounce'
+export * from './useDebouncePair'
 export * from './useEtherBalance'
-export * from './useToken'
-export * from './useTokenBalance'
-export * from './useTokenAllowance'
+export * from './useEthers'
+export * from './useGasPrice'
```
Comment (DoubleOTheven) [inline packages/core/src/hooks/index.ts:10]: I just sorted everything alphabetically. No changes...

**Comment by DoubleOTheven** [comment]
Thoughts on this? @marekkirejczyk

**Comment by marekkirejczyk** [comment]
Hi @DoubleOTheven,

It is a great idea! Love your work ❤️ 

We are working however now on a series of cross-dependent features and refactorings, including:
- remove web3react dependency #491
- add example for Web3Dialog (draft in #489)
- enable multichain support

We will probably have new hook `useNetworks()` that will fulfil your needs.
For now we want to wait and see how the refactoring will shape out.

**Comment by DoubleOTheven** [comment]
> Hi @DoubleOTheven,
> 
> It is a great idea! Love your work ❤️
> 
> We are working however now on a series of cross-dependent features and refactorings, including:
> 
> * remove web3react dependency [🚮 Remove web3-react dependency from core #491](https://github.com/EthWorks/useDApp/pull/491)
> * add example for Web3Dialog (draft in [🥶 Bump web3-react to 8.0.0-beta, add Web3Modal to example (#479) #489](https://github.com/EthWorks/useDApp/pull/489))
> * enable multichain support
> 
> We will probably have new hook `useNetworks()` that will fulfil your needs. For now we want to wait and see how the refactoring will shape out.

Sounds good. I'll close this one then.

Can we start a dev chat somewhere? It would be good so I can stay in the loop and help contribute / know what is being planned, etc Can you DM me in Twitter? `@SamRuberti`

**Comment by vanruch** [comment]
@DoubleOTheven You can join https://discord.com/invite/ppFxC3E44X #usedapp-support channel for now

**Pull request closed without merging**